### PR TITLE
Kafka-Python Serialization Metrics

### PR DESCRIPTION
**Pull request by TimPansino**
# Overview
* Adds serialization metrics for kafka-python with thorough testing.
* Does NOT include deserialization metrics, as the implementation turned out to be too difficult for the amount of value it would provide. If this is a hotly requested feature it may be revisited.

### CODE REVIEW

File: tests/messagebroker_kafkapython/test_consumer.py [hunk 1]
```diff
@@ -12,15 +12,20 @@
...
+
 from testing_support.fixtures import (
     validate_attributes,
     validate_error_event_attributes_outside_transaction,
     validate_transaction_errors,
     validate_transaction_metrics,
+    reset_core_stats_engine,
+)
+from testing_support.validators.validate_transaction_count import (
+    validate_transaction_count,
```
Comment (hmstepanek) [inline tests/messagebroker_kafkapython/test_consumer.py:28]: Let's add the following extra test in here:
```
def test_multiple_transactions(get_consumer_record, topic):
    @validate_transaction_count(2)
    def _test():
        get_consumer_record()
        get_consumer_record()

    _test()
```

**Review by hmstepanek — CHANGES_REQUESTED** [review]
I think we should add that test in that I mentioned earlier today about testing multiple transactions back to back for a single consumer. I also had a concern about adding the children loop without being under an if terminal.

File: newrelic/core/message_node.py [hunk 2]
```diff
@@ -51,6 +51,12 @@ def time_metrics(self, stats, root, parent):
         yield TimeMetric(name=name, scope=root.path,
                 duration=self.duration, exclusive=self.exclusive)
 
+        # Now for the children, if the trace is not terminal.
```
[the remark this answers is not in the document]
Comment (TimPansino) [reply newrelic/core/message_node.py:54]: In the __enter__ of TimeTrace, traces will refuse to start if the parent is a terminal node. Which is what we observed while testing this.

I'm fine with putting a guard in anyway but it's probably redundant.

[the remark this answers is not in the document]
Comment (TimPansino) [reply newrelic/core/message_node.py:54]: Actually we can't do that at all. The MessageNode object doesn't have any access to the terminal attribute or the trace the created it and has no idea if it was terminal.

This is also true of FunctionTraces which are also configurable to be terminal or not. They don't have any guard for this, and don't preserve any means to tell if they were terminal or not.

So it seems the `TimeTrace.__enter__` guard is sufficient.

[the remark this answers is not in the document]
Comment (hmstepanek) [reply newrelic/core/message_node.py:54]: Ah ok I completely forgot about the __enter__ catching it. Ok that works then!

**Pull request merged**